### PR TITLE
LG-2937: bin/activate will download application.yml from S3 and merge in

### PR DESCRIPTION
**Pull request by solipet**
In looking to mirror the way the IDP works, I finally took a look at the application.yml's that are up on `dev` and `int` and I'd like to start working on cleaning them up.

I think about 40% of the settings could just live in `application.yml.example` since they aren't secret or different across the environments. I included these settings in `application.yml.example` directly.

The configs stored up on S3 currently have all settings in alphabetical order. I'd like to remove the common settings I've included in the example config, then list the remainder in alphabetical order, but broken out into a few groups: feature flags, secrets, overrides, etc.

Finally, in the example config, we have settings for `saml_sp_certificate`, `saml_sp_private_key`, and `saml_sp_private_key_password`. These are listed as required in `config/initializers/figaro.rb`, but are not set in either the `dev` or `int` application.yml's. The certificate is not referenced anywhere in the codebase. The key and password are used in the omniauth config, but that's an OIDC config, so...

### CODE REVIEW

File: lib/deploy/activate.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        '/%<env>s/dashboard/v1/application.yml' => env_yaml_path
+      )
+    end
+
+    def deep_merge_s3_data_with_example_application_yml
+      File.open(result_yaml_path, 'w') { |file| file.puts YAML.dump(application_config) }
+    end
+
+    def set_proper_file_permissions_for_application_yml
+      FileUtils.chmod(0o640, [env_yaml_path, result_yaml_path])
```
Comment (zachmargolis) [inline lib/deploy/activate.rb:35]: oooh fancy octal notation, I didn't realize this was in the IDP too

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -12,6 +12,7 @@
 # http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
 RSpec.configure do |config|
   config.color = true
+  config.formatter = :documentation
```
Comment (zachmargolis) [inline spec/spec_helper.rb:15]: 😍

File: config/application.yml.example [hunk 3]
```diff
@@ -1,3 +1,44 @@
...
+disallow_ial2_recovery: 'false'
+doc_auth_enabled: 'true'
+doc_auth_exclusive: 'true'
+email_deletion_enabled: 'true'
+email_from: no-reply@identitysandbox.gov
+enable_usps_verification: 'true'
+enable_load_testing_mode: 'true'
+enable_mobile_capture: 'true'
+hmac_fingerprinter_key_queue: '["old-key-one", "old-key-two"]'
+in_person_proofing_enabled: 'false'
```
Comment (zachmargolis) [inline config/application.yml.example:24]: this is not a dashboard feature flag as far as I can tell....should we remove it?

File: config/application.yml.example [hunk 4]
```diff
@@ -1,3 +1,44 @@
...
+aamva_verification_request_timeout: '50'
+aamva_verification_url: https://verificationservices-cert.aamva.org:18449/dldv/2.1/online
+account_reset_wait_period_days: '0'
+acuant_timeout: '55'
+acuant_simulator: 'true'
+acuant_sdk_document_capture_enabled: 'true'
+allow_doc_auth_test_credentials: 'true'
+attribute_encryption_key_queue: '["old-key-one", "old-key-two"]'
+aws_region: us-west-2
+cac_proofing_enabled: 'true'
```
Comment (zachmargolis) [inline config/application.yml.example:12]: ditto .... this is an IDP feature flag, I think we should remove it here

Comment (jmhooper) [reply config/application.yml.example:12]: Yeah, I think there are quite a few IdP feature flags in here 🤔

Comment (solipet) [reply config/application.yml.example:12]: Ugh.. I think I mixed up the idp application.yml!! 🤦

**Comment by solipet** [comment]
@zachmargolis @jmhooper I've cleaned this up and uploaded the current `application.yml` in use on dev/int to s3 in preparation of this becoming the new way.

**Comment by solipet** [comment]
> LGTM, looks like a spec failure, but once that get resolved go ahead and merge:
> 
> ```
>      # ./spec/features/logo_upload_spec.rb:26:in `block (3 levels) in <top (required)>'
> ```
> 
> Maybe related to `fakefs` messing with filesystem access?

I really don't understand what changed between the passing test and now that could cause a consistent failure.  `git diff c877402 78f5356`. I cleaned up the example yml file. I moved `smtp.rb` from `config/` to `config/initializers/` and told `production.rb` where to find it. I removed the requirement to have `saml_sp_certificate` in `figaro.rb` since we no longer use it. And finally, I updated the `activate_spec` to look for settings that are actually in the example yml now. (╯ಠ益ಠ)╯彡 ┻━┻

**Comment by solipet** [comment]
> > LGTM, looks like a spec failure, but once that get resolved go ahead and merge:
> > ```
> >      # ./spec/features/logo_upload_spec.rb:26:in `block (3 levels) in <top (required)>'
> > ```
> > 
> > 
> > Maybe related to `fakefs` messing with filesystem access?
> 
> I really don't understand what changed between the passing test and now that could cause a consistent failure. `git diff c877402 78f5356`. I cleaned up the example yml file. I moved `smtp.rb` from `config/` to `config/initializers/` and told `production.rb` where to find it. I removed the requirement to have `saml_sp_certificate` in `figaro.rb` since we no longer use it. And finally, I updated the `activate_spec` to look for settings that are actually in the example yml now. (╯ಠ益ಠ)╯彡 ┻━┻

Bah!, nevermind -- found it 😝

**Pull request merged**